jax/_src/lax/lax.py

```diff
@@ -6652,28 +6652,30 @@ def remaining(original, *removed_lists):
   return [i for i in original if i not in removed]
 
 
-def _canonicalize_precision(precision):
+def _canonicalize_precision(precision: PrecisionLike) -> Optional[Tuple[PrecisionType, PrecisionType]]:
```

Comment:
Doc string? (e.g. to explain the meaning of the tuple values)

Reply:
I added a docstring. I wish we did have more docstrings. For now, what we are trying to do is to have docstrings for all API functions, then ideally for all module exports, and probably not add them for module locals. I think that we should
add type annotations for all functions though.